lib/multi_fetch_fragments.rb

```diff
@@ -68,6 +75,9 @@ def render_collection_with_multi_fetch_cache
             results << cached_value
           else
             non_cached_result = non_cached_results.shift
+            if !@expires_collection.empty?
```

Comment:
One more thing - you may want to do a `merge!` instead, as you may run the risk of blowing out the existing values in the existing Hash. So perhaps the right line to use is here:

```ruby
additional_cache_options.merge!(:expires_in => @expires_collection[key] if @expires_collection.has_key?(key)
```